### PR TITLE
Support torch patch base images

### PR DESCRIPTION
**Pull request by 8W9aG**
* Do not strip patch version from torch
* If the user specifies a torch version without a patch, resolve the highest patch version from our compatibility matrix
* Change the version struct to use a pointer for the patch version in order to handle cases where the pointer has not been supplied (and how that manifests when comparing versions)
* Standardise on StripModifier for the version rendering
* No longer generate a torch minor compatibility matrix, unnecessary now we explicitly define patch version compatibilities

### CODE REVIEW

File: pkg/dockerfile/base.go [hunk 1]
```diff
@@ -110,17 +108,19 @@ func BaseImageConfigurations() []BaseImageConfiguration {
...
 
 	// Torch configs
-	for _, compat := range config.TorchMinorCompatibilityMatrix {
+	for _, compat := range config.TorchCompatibilityMatrix {
 		for _, python := range compat.Pythons {
 
-			// Only support fast cold boots for Torch with CUDA.
-			// Torch without CUDA is a rarely used edge case.
 			if compat.CUDA == nil {
+				configs = append(configs, BaseImageConfiguration{
```
Comment (andreasjansson) [inline pkg/dockerfile/base.go:115]: Is there a reason to build base images for torch without gpu? Feels like a very edgy edge case.

File: pkg/util/version/version.go [hunk 2]
```diff
@@ -32,7 +32,9 @@ func NewVersion(s string) (version *Version, err error) {
 		}
 	}
 	if len(parts) >= 3 {
-		version.Patch, err = strconv.Atoi(parts[2])
+		patch, err := strconv.Atoi(parts[2])
+		version.Patch = new(int)
```
Comment (andreasjansson) [inline pkg/util/version/version.go:36]: Why is this necessary? Warrants a comment. And should this happen after the error check?

**Review by andreasjansson — COMMENTED** [review]
We should also stop stripping out torch and torchvision from the generated requirements.txt in the build process, but that's probably a different PR?

**Comment by 8W9aG** [comment]
> We should also stop stripping out torch and torchvision from the generated requirements.txt in the build process, but that's probably a different PR?

Yeh the next PR after this one handles torchvision and torchaudio

File: pkg/util/version/version.go (hunk 2 above)
Comment (8W9aG) [reply pkg/util/version/version.go:36]: The error check should be before the Patch pointer allocation you are right, I'll add a comment.

File: pkg/dockerfile/base.go (hunk 1 above)
Comment (andreasjansson) [reply pkg/dockerfile/base.go:115]: I see, yes that makes sense.
